src/components/user/user.repository.ts

```diff
@@ -243,6 +244,30 @@ export class UserRepository extends DtoRepository<typeof User, [Session | ID]>(
     return !!result;
   }
 
+  async getUserByEmailAddress(email: string, session: Session) {
+    const query = this.db
+      .query()
+      .matchNode('node', 'User')
+      .where(
+        path([
+          node('node'),
+          relation('out', '', 'email', ACTIVE),
+          node({
+            value: email,
+          }),
+        ]),
+      )
```

Comment:
I think we also need to `filterToReadable` here. We don't want anyone to be able to probe users by public email addresses